Refactored more docstrings in :mod:`.geometry`

Kiran-Raj-Dev opened the pull request:
## Overview: What does this pull request change?
Changes the order of sections within docstrings to follow the numpy format as per the [documentation](https://docs.manim.community/en/latest/contributing/docstrings.html)
## Links to added or changed documentation pages
Furthur changes at
https://manimce--2560.org.readthedocs.build/en/2560/reference/manim.mobject.geometry.arc.html
https://manimce--2560.org.readthedocs.build/en/2560/reference/manim.mobject.geometry.boolean_ops.html
https://manimce--2560.org.readthedocs.build/en/2560/reference/manim.mobject.geometry.line.html#

<!-- Thank you again for contributing! Do not modify the lines below, they are for reviewers. -->
## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

icedcoffeeee reviewed (approved):
> If we could come to a conclusion of the format, then not only does that ensure consistency but also hopefully prevent such bugs from occuring.

I think the format is already concluded in the [Adding Docstrings](https://docs.manim.community/en/latest/contributing/docstrings.html) part of the documentation. I guess reviewers, such as myself, don't look too deep into these things.

> Most places in the docs suffer from inconsistencies, in how the sections (like Parameters, Return type, Examples) appear

Are there any more places where these issues happen? If so, it might be best to fix them in a single PR.

If not, this looks good to me!

Kiran-Raj-Dev commented:
> I think the format is already concluded in the [Adding Docstrings](https://docs.manim.community/en/latest/contributing/docstrings.html) part of the documentation. I guess reviewers, such as myself, don't look too deep into these things.

Sorry, I didn't look into that part of the documentation before making that comment on docstring format.
So, numpy suggests,

1. Summary
2. Extended summary
3. Parameters / Methods / Attributes
4. Returns
4. See also
5. Notes
6. References
7. Examples

And taking a look at [Circle](https://docs.manim.community/en/stable/reference/manim.mobject.geometry.Circle.html) for example that doesn't seem to be the case.

I'll try to fix as much of them as I can. Until then I'll mark this PR as a draft.

Kiran-Raj-Dev commented:
Also, in some places we have separated `Returns` and `Return Type`.
Numpy suggests to have the `Return Type` mentioned within the `Returns` section.
Thought?

icedcoffeeee commented:
> I'll try to fix as much of them as I can.

I was actually referring to the issues of duplicated parameters, rather than the numpy formatting. If there aren't any clear issues (as with `Vector.coordinate_label`), it might not be worth the effort to rewrite.

> Also, in some places we have separated Returns and Return Type. Numpy suggests to have the Return Type mentioned within the Returns section.

In future PR reviews, I think following the latter is better.

Kiran-Raj-Dev commented:
> I was actually referring to the issues of duplicated parameters, rather than the numpy formatting. If there aren't any clear issues (as with Vector.coordinate_label), it might not be worth the effort to rewrite.

Well, I already did some 😅.
Should I continue or do we stop midway?

icedcoffeeee commented:
> Well, I already did some 😅. Should I continue or do we stop midway?

Welp, I guess that's fine then 👍

Kiran-Raj-Dev commented:
No this is messed up again 😢

Kiran-Raj-Dev commented:
I think I'll only change mobjects and animations since those are frequented by people and also the ones with a lot of examples and docstrings.

icedcoffeeee reviewed (changes requested):
Multiple admonition notes are broken: ArcPolygon, Cutout, Graph, just to name a few, i wonder if these rearrangements caused it? 🤔

Kiran-Raj-Dev commented:
Oh no.
Looks like my editor broke some indentation levels when re-arranging.

Shall fix it right away

Kiran-Raj-Dev commented:
Well, it wasn't the rearrangement. (maybe?)
Apparently sphinx has automatically added the parameters that weren't documented, breaking the admonitions

Kiran-Raj-Dev commented:
Classic example of the library trying to be smarter than the user

Kiran-Raj-Dev commented:
Some more changes that I've made.
- In order to tell sphinx that parameters should be forst priority I have added in *dummy* parameters like:
```
Parameters
-----------
<the first __init__ argument>
```
- I have added docstrings to places where they weren't present before
- I have added some decriptions to places where it was easy enough. For example, what is the `start` parameter in the function `put_start_and_end_on`
- Imported VMobject in movement.py for typechecking

I have gone through all of the classes and checked for broken admonitions and fixed them.

Kiran-Raj-Dev commented:
Oh and adding an example for the `align_to` method broke the docs build for some reason.

behackl commented:
> Oh and adding an example for the `align_to` method broke the docs build for some reason.

The way it is formatted in the comment it is missing a newline between the directive and the code. I can say more if you find me the logs of the failed build.

behackl reviewed (commented):
Thank you for your efforts! Some notes: this will cause (more) merge conflicts in @hydrobeam's module refactor PR #2476 -- which I'd actually like to get merged first, except if @hydrobeam doesn't think it would complicate the situation even more. 

Some section headings don't have the correct amount of dashes (but also didn't do so before); I'd prefer if they were correct but I won't block merging this over it.

What I genuinely do not like here is the added partial Parameters-sections. Can you explain why you think they should be added? Also, in terms of consistency I think the detail descriptions of keyword arguments are usually ended with a full stop (this is from memory though and could be wrong).

In terms of review-difficulty, one PR per changed module also would have been better.

Kiran-Raj-Dev commented:
> Some section headings don't have the correct amount of dashes (but also didn't do so before); I'd prefer if they were correct.

Hmm, I think flake8 could do that automatically. Will look into it.

> What I genuinely do not like here is the added partial Parameters-sections. Can you explain why you think they should be added?

Well that was one way of telling sphinx to add the parameters first at the time of making this PR. I did some digging in the sphinx documentation, but couldn't find any. I'll continue looking into the docs if this approach doesn't look nice. And if you come across some then let me know :)

> Also, in terms of consistency I think the detail descriptions of keyword arguments are usually ended with a full stop 

Shall add them.

> In terms of review-difficulty, one PR per changed module also would have been better.

There is a one commit per changed module though :)
But then again just two commit for the admonition changes.

behackl commented:
> There is a one commit per changed module though :)

This is indeed better than no separation at all, but not entirely the point why I think more smaller PRs would be better: if there is a problem / discussion for some docstrings in one module, all other modules that would otherwise be fine are blocked from being merged as well. And additionally, a PR with ±1k changed lines takes a substantial amount of time to get through and give feedback on.

Kiran-Raj-Dev commented:
So, what should be the course of action now?

Darylgolden commented:
I suppose you could cherry pick commits and split them off into different PRs.

Kiran-Raj-Dev commented:
Made changes to line, boolean ops and arc.
The rest will go in a new PR.

behackl commented on manim/mobject/geometry/arc.py:
```diff
@@ -641,11 +640,11 @@ def construct(self):
     def __init__(
         self,
-        point=ORIGIN,
+        point: list | np.ndarray | None = ORIGIN,
```
The optional annotation was wrong, point cannot actually be `None`.

```suggestion
        point: list | np.ndarray = ORIGIN,
```

behackl reviewed (approved):
Two minor remarks, please have a look. Otherwise LGTM, thank you!

Kiran-Raj-Dev commented on docs/source/conf.py:
```diff
@@ -71,7 +71,7 @@
 html_extra_path = ["robots.txt"]
 
-exclude_patterns: list = []
+exclude_patterns: list[str] = ["reference/manim.mobject.opengl**"]
```
On my machine sphinx was building the docs for OpenGL as well, which failed due to `OpenglVGroup` not being defined (I have no idea why that was happening) so excluded them from the build using this switch and it worked after that.

Docs [here](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-exclude_patterns)

Darylgolden replied:
Does it fail on RTD? If not I think we can undo this change and you should file an issue. And even if it does fail, it's better to split it up into a new PR for cleaner commit history, since this seems unrelated.

Kiran-Raj-Dev replied:
Ok shall revert

Darylgolden reviewed (changes requested):
Left a comment.

icedcoffeeee reviewed (approved):
LGTM
